### DISCUSSION TITLE
Laravel Zero and NativePHP?

### DISCUSSION THREAD

**Discussion by osteel** [laravel-zero/laravel-zero#457] [Ideas]
Hi there,

[NativePHP](https://nativephp.com/) was recently unveiled at Laracon and I'm pretty excited about it. What would get me even more excited though is being able to distribute Laravel Zero applications as native programs, whether through NativePHP or not.

I'm wondering how hard it would be to get the ball rolling on this? Has anyone started looking into it yet?

I know @mpociot [mentioned submitting a PR for this](https://twitter.com/marcelpociot/status/1676585881795272709) (also suggesting bundling the PHP binary with the Laravel Zero app wouldn't be done through NativePHP), but I'm guessing that's not very high on his priority list.

I'd be happy to help, but have no idea where to start at the moment. Is there any interest from Laravel Zero to get something like that in, to begin with? Would [Static PHP CLI](https://github.com/crazywhalecc/static-php-cli/) be the right starting point for it?

Any pointers would be appreciated!

Cheers,

Yannick

**Comment by mpociot** [comment]
I actually started working on this already, I hope to finish the PR over the weekend 👍

**Reply by osteel** [reply]
Now I am _very_ excited.

**Reply by mrfelipemartins** [reply]
Awesome! Can't wait to see this PR!

**Reply by osteel** [reply]
haha yeah same – I'm checking once in a while, see if it pops up

**Reply by igorsgm** [reply]
I'm very excited for it too! 👀

**Comment by emmadesilva** [comment]
I think the only thing that's really needed is some way to bundle the static PHP binary?

**Reply by osteel** [reply]
Basically yes. Would just be nice to have the corresponding logic abstracted away behind a nice API of some sorts.

I also have [FrankenPHP](https://frankenphp.dev/docs/embed/) on my radar but haven't had the chance to give it a go yet

**Comment by osteel** [comment]
Quick update on this – while some sort of wrapper with a nice API would be cool, it's probably possible to package a Laravel Zero app with a PHP runtime already.

[This article](https://www.bosunegberinde.com/articles/building-php-binary) is a good starting point and the example application the author uses actually works.

I tried to package a Laravel Zero project [of mine](https://github.com/osteel/dime) the same way but it unfortunately didn't work. I keep getting the following error and I don't know how to debug it at this stage:

```
The application may only be invoked from a command line, got "micro"
```

Others might have more luck though!

**EDIT:** Just got the solution from the article's author [on X](https://x.com/bosunski/status/1750881775092462050?s=20) – there's a `--with-micro-fake-cli` [option](https://static-php.dev/en/guide/manual-build.html#build-options) when building the static PHP binary that addresses this error specifically
